sql/core/src/test/scala/org/apache/spark/sql/StatisticsSuite.scala

```diff
@@ -101,4 +101,47 @@ class StatisticsSuite extends QueryTest with SharedSQLContext {
       checkTableStats(tableName, expectedRowCount = Some(2))
     }
   }
+
+  test("test column-level statistics for data source table created in InMemoryCatalog") {
+    def checkColStats(colStats: BasicColStats, expectedColStats: BasicColStats): Unit = {
+      assert(colStats.dataType == expectedColStats.dataType)
+      assert(colStats.numNulls == expectedColStats.numNulls)
+      assert(colStats.max == expectedColStats.max)
+      assert(colStats.min == expectedColStats.min)
+      if (expectedColStats.ndv.isDefined) {
+        // ndv is an approximate value, so we just make sure we have the value
+        assert(colStats.ndv.get >= 0)
```

Comment:
Because the check here for estimated ndv value is not really related to the actual ndv value, we can't make sure if we really get the estimated ndv back. Especially we only use `ndv > 0` to check it. In an extreme case, we can replace the ndv expression with a literal(0) in the aggregation, the test would not check it out.